src/com/rcv/CVRReader.java

```diff
@@ -23,162 +23,197 @@
 
 class CVRReader {
 
-  // container for all CastVoteRecords parsed from the input file
-  final List<CastVoteRecord> castVoteRecords = new ArrayList<>();
-
-  // purpose: helper function to wrap file IO with error handling
-  // param: excelFilePath path to file for parsing
-  // file access: read
-  // throws: IOException if there was a problem opening or reading the file
-  // returns: the first xls sheet object in the file or null if there was a problem
-  private static Sheet getFirstSheet(String excelFilePath) throws IOException {
-    // container for result
-    Sheet firstSheet;
-    try {
-      // inputStream for parsing file data into memory
-      FileInputStream inputStream = new FileInputStream(new File(excelFilePath));
-      // excel workbook object to access to sheet objects it contains
-      Workbook workbook = new XSSFWorkbook(inputStream);
-      firstSheet = workbook.getSheetAt(0);
-      inputStream.close();
-      workbook.close();
-    } catch (IOException exception) {
-      Logger.severe("failed to open CVR file: %s, %s", excelFilePath, exception.getMessage());
-      throw exception;
-    }
-    return firstSheet;
-  }
-
-  // purpose: parse the given file path into a CastVoteRecordList suitable for tabulation
-  // Note: this is specific for the Maine example file we were provided
+  // config for the election
+  private ElectionConfig config;
+  // path of the source file
+  private String excelFilePath;
+  // column index of first ranking
+  private int firstVoteColumnIndex;
+  // column index of CVR ID (if present)
+  private Integer idColumnIndex;
```

Comment:
I use the wrapper class instead of the primitive when I want the variable to be nullable. Since Java does auto-unboxing of these wrapper objects, it's safe to treat them as if they were primitives when you know that they're not null.